### ISSUE TITLE
IDE support for e2e tests

### ISSUE THREAD

**Issue by oreissig** [operator-framework/operator-sdk#1479]
## Type of question

**Are you asking about community best practices, how to implement a specific feature, or about general context and help around the operator-sdk?**

Asking for prior experience or best practices, results should then ideally be persisted in `writing-e2e-tests.md`.

## Question

**What did you do?**
**A clear and concise description of the steps you took (or insert a code snippet).**

I followed `writing-e2e-tests.md` and am now able to run `operator-sdk test local ./test/e2e --namespace operator-test --up-local`

**What did you expect to see?**
I'd like to be able to use my IDE to run and debug both the operator itself, as well as the test runner massaging my Kubernetes cluster.

**What did you see instead? Under which circumstances?**
This is currently cumbersome to iterate on in case something fails. You can't even get the operator's logs when not using `--up-local`, giving no indication of what may have caused the problem.

**Environment**
* operator-sdk version:

0.8.0

* Kubernetes version information:

Using the Kubernetes built into Docker for Mac.
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T18:55:03Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

it is very kind :-)

**Additional context**
Using IntelliJ, but VSCode would be okay as well.

**Comment by LiliC** [comment]
Not sure if this is useful but there is [this blog post](https://dev4devs.com/2019/05/04/operator-framework-how-to-debug-golang-operator-projects/) by @camilamacedo86 about how to debug operators. Let me know if that is of any help!

**Comment by AlexNPavel** [comment]
To add onto what @LiliC said, there is currently a PR out to add delve debugging support to the `up local` subcommand, similar to what is mentioned in the above linked blogpost. Once this PR is merged, if you have delve installed you can simply do `operator-sdk up local --enable-delve` and then point your IDE's debugger to local port `:2345`: https://github.com/operator-framework/operator-sdk/pull/1422

**Comment by oreissig** [comment]
>  there is currently a PR out to add delve debugging support to the `up local` subcommand

That sounds awesome!
I'm still having a hard time groking the `writing-e2e-tests.md`, so providing more assistence about the practical flow of working with e2e tests (incl. "how do I debug in case a test fails") in this guide would be greatly appreciated.

**Comment by LiliC** [comment]
@oreissig since https://github.com/operator-framework/operator-sdk/pull/1422 was merged, does this solve your issue and we can close this? :)

**Comment by oreissig** [comment]
@LiliC hmm actually not really. It's only part of the solution, but I'm still lacking a convenient workflow that would allow me to iterate on my code in a test-driven way (beyond what can be done with unit tests and fake client).

I feel overwhelmed by the content presented in the `writing-e2e-tests.md` document. Some things that come to my mind:
- It references some sample test case, that I need to manually copy&paste. Maybe it should just pop out of the scaffolding process?
- The "Walkthrough: Writing Tests" part of `writing-e2e-tests.md` feels disconnected, maybe this should instead be comments inside the (generated) test code?
- How I can use the newly added debugging capability should be mentioned somewhere in `writing-e2e-tests.md`.
- I have found there to be quite a difference between the controller runtime's fake client and the behaviour of an actual Kubernetes cluster. Having the possibility to run the actual operator directly from the IDE without it being embedded in a Docker container may be a good balance between "integratedness" and feedback loop cycle time?

Maybe we don't actually need to provide a new mechanism for anything, but just add some more documentation? I'm not sure what the best approach is in the first place.

**Comment by LiliC** [comment]
@oreissig Most of these sound like unrelated to your original issue which is to support IDE, can you open a separate issue for the above ones, it's more about updating docs and possibly generating some e2e scaffold code, thanks!

> I have found there to be quite a difference between the controller runtime's fake client and the behaviour of an actual Kubernetes cluster. Having the possibility to run the actual operator directly from the IDE without it being embedded in a Docker container may be a good balance between "integratedness" and feedback loop cycle time?

You are using the fake client? In the e2e tests we have we tend to run those against a real k8s cluster, the fake client is more used for unit tests.

**Comment by oreissig** [comment]
I think I realized that
1. I'm not able to provide clear requirements of where I'd like this to be taken.
2. Others don't seem to have the same uncertainties, or at least they don't engage in this conversation to add some substance.
With that there's no really constructive way to take this ticket forward.
